ibis/pandas/core.py

```diff
@@ -1,3 +1,80 @@
+"""The pandas backend is a departure from the typical ibis backend in that it
+doesn't compile to anything, and the execution of the ibis expression coming
+dervied from a PandasTable is under the purview of ibis itself rather than
+sending something to a server and having it return the results.
+
+Design
+------
+
+Ibis uses a technique called `multiple dispatch
+<https://en.wikipedia.org/wiki/Multiple_dispatch>`_, implemented in a
+third-party open source library called `multipledispatch
+<https://github.com/mrocklin/multipledispatch>`_.
+
+Multiple dispatch is a generalization of standard single-dispatch runtime
+polymorphism to multiple arguments.
+
+Compilation
+-----------
+This is a no-op because we directly execute ibis expressions.
+
+Execution
+---------
+
+Execution is divided into different dispatched functions, each arising from
+different a use case.
+
+A top level dispatched function named ``execute`` with two signatures exists
+to provide a single API for executing an ibis expression.
+
+The general flow of execution is:
+
+::
+       If the current operation is in scope:
+           return it
+       Else:
+           execute the arguments of the current node
+
+       execute the current node with its executed arguments
+
```

Comment:
I think this rather belongs in the docs proper

Reply:
So, I put this here so that I could start a trend of adding module level docs for each backend, rather than have one rst file for each backend which will tend to get out of date less often than the module docs.